hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/rollback/BaseRollbackActionExecutor.java

```diff
@@ -297,4 +301,35 @@ protected void dropBootstrapIndexIfNeeded(HoodieInstant instantToRollback) {
       BootstrapIndex.getBootstrapIndex(table.getMetaClient()).dropIndex();
     }
   }
+
+  private void backupRollbackInstantsIfNeeded() {
+    if (!config.shouldBackupRollbacks()) {
+      // Backup not required
+      return;
+    }
+
+    Path backupDir = new Path(config.getRollbackBackupDirectory());
+    if (!backupDir.isAbsolute()) {
+      // Path specified is relative to the meta directory
+      backupDir = new Path(table.getMetaClient().getMetaPath(), config.getRollbackBackupDirectory());
+    }
+
+    // Determine the instants to backup. Completed and inflight instants are backed up.
+    HoodieActiveTimeline activeTimeline = table.getActiveTimeline();
+    List<HoodieInstant> instantsToBackup = new ArrayList<HoodieInstant>(3);
+    instantsToBackup.add(instantToRollback);
```

Comment:
Some question:

1. why we also back up the inflight instant when the instant is complete?
2. what kind of info from the instant is useful for debugging from your side, can we make it more light-weight?

Reply:
this is inline w/ what we do for archival. we will retain entire history. also helps w/ investigation w/ metadata going out of sync may be.